internal/core/domain/market_service.go

```diff
@@ -52,10 +40,8 @@ func (m *Market) QuoteAssetPrice() decimal.Decimal {
 }
 
 // IsStrategyPluggable returns true if the the startegy isn't automated.
-// For backward compatibility it returns whether the strategy is zero-ed or,
-// on the contrary, its type is extaclty PluggableStrategyType.
 func (m *Market) IsStrategyPluggable() bool {
-	return m.Strategy.IsZero() || m.Strategy.Type == PluggableStrategyType
+	return !m.Strategy.IsZero() && m.Strategy.Type == int(StrategyTypePluggable)
```

Comment:
what is the impact of this change?

Reply:
Since the type defined for the Pluggable strategy in the proto  is `0`, we need to be sure that in case the market strategy type has this kind of value, its formula must not be nil because it would mean that no strategy has been set for the current market. This is the impact of the change.

Before this, since `IsZero()` was just checking the type and not also the formula, the workaround was to use the `255` value for the pluggable strategy type (which generated #500).

The other change that contributes to fixing that ticket is [this](https://github.com/altafan/tdex-daemon/blob/fix-fixed-fee-and-strategy/internal/infrastructure/storage/db/badger/market_repository_impl.go#L366): before this, in the badger repo we were restoring only the non-pluggable strategies, this was due to the fact that in the early versions, Pluggable was not an actual `MakingFormula` implementation but rather, all its logic was hardcoded in the application level.
Now, instead, we restore all kinds of strategies, and evenmore, using the `default` branch for the pluggable one makes it compatible with markets storing either the `255` or the `0` type.
The very first change to the market, like for example a change of the price by some feeder, will then make the pluggable strategy type to be set to the correct `0` value for that market.